### PR TITLE
Fix hash_balance_data

### LINKED ISSUE #1975: hash_balance_data returns the wrong balance hash
## Problem Definition

the function [`hash_balance_data`](https://github.com/raiden-network/raiden/blob/master/raiden/transfer/utils.py#L6) returns the wrong balance hash. 
Example:
```
>>> hash_balance_data(0, 0, b'')                                                                                                                                                                                                                                                
HexBytes('0xad3228b676f7d3cd4284a5443f17f1962b36e491b30a40b2405849e597ba5fb5')
```
In this case, the balance hash return value should be `0x0000000000000000000000000000000000000000000000000000000000000000`. 
This might become an issue given that the logic implemented in the [contracts](https://github.com/raiden-network/raiden-contracts/blob/master/raiden_contracts/contracts/TokenNetwork.sol#L1013) doesn't use the balance hash if `transferred_amount`, `locked_amount` and `locksroot` are all zeroed out.

### PR DESCRIPTION
**Pull request by rakanalh**
Resolves #1975 

TODO:
- [x] Add changelog entry
- [x] Fix tests

### CODE REVIEW

File: raiden/tests/unit/transfer/test_utils.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    test_table = {
+        (0, 0, b''): decode_hex(
+            '0x0000000000000000000000000000000000000000000000000000000000000000',
+        ),
+        (1, 5, EMPTY_MERKLE_ROOT): decode_hex(
+            '0xc6b26a4554afa01fb3409b3bd6e7605a1c1af45b7e644282c6ebf34eddb6f893',
+        ),
+    }
+
+    for values, result in test_table.items():
```
Comment (palango) [inline raiden/tests/unit/transfer/test_utils.py]: nitpick: you could use pytest for this: https://github.com/ethereum/eth-utils/blob/master/tests/hexadecimal-utils/test_is_0x_prefixed.py#L8

File: raiden/transfer/utils.py [hunk 2]
```diff
@@ -8,6 +8,10 @@ def hash_balance_data(
         locked_amount: typing.TokenAmount,
         locksroot: typing.Locksroot,
 ) -> str:
+    if all([transferred_amount == 0, locked_amount == 0, locksroot == b'']):
```
Comment (hackaugusto) [inline raiden/transfer/utils.py]: Please just use `transferred_amount == 0 and locked_amount == 0 and locksroot == b''`

`all([])` will create a temporary list and do a function call for something that is supported in the language.

File: raiden/transfer/utils.py [hunk 3]
```diff
@@ -8,6 +8,10 @@ def hash_balance_data(
...
+        return decode_hex(
+            '0x0000000000000000000000000000000000000000000000000000000000000000',
```
Comment (hackaugusto) [inline raiden/transfer/utils.py]: this could be a constant

File: raiden/transfer/utils.py [hunk 4]
```diff
@@ -8,6 +8,10 @@ def hash_balance_data(
         locked_amount: typing.TokenAmount,
         locksroot: typing.Locksroot,
 ) -> str:
+    if all([transferred_amount == 0, locked_amount == 0, locksroot == b'']):
+        return decode_hex(
```
Comment (pcppcp) [inline raiden/transfer/utils.py]: `return bytes(32)` ?

File: raiden/tests/unit/transfer/test_utils.py (hunk 1 above)
Comment (hackaugusto) [reply raiden/tests/unit/transfer/test_utils.py]: side note though: this feature is really slow, don't use it with a large list (for this is fine)

**Comment by LefterisJP** [comment]
@rakanalh Can you also add a changelog entry?

**Comment by rakanalh** [comment]
:heavy_check_mark:  Integration test build: https://travis-ci.org/raiden-network/raiden/builds/408917007?utm_source=github_status&utm_medium=notification

File: docs/changelog.rst [hunk 5]
```diff
@@ -11,7 +11,7 @@ Changelog
 * :bug:`1902` Check for ethnode connection at start and print proper error if Raiden can not connect
 * :bug:`1911` The syncing message is now printed properly and does not repeat across the screen
 * :bug:`1899` Print proper error without throwing exception if no accounts are found in the keystore
-
+* :bug:`1975` Fix balance hash generation to match the logic that lives in ``TokenNetwork`` contract.
```
Comment (palango) [inline docs/changelog.rst]: Would be good to have a new line after this.

File: docs/changelog.rst [hunk 6]
```diff
@@ -11,9 +11,9 @@ Changelog
 * :bug:`1902` Check for ethnode connection at start and print proper error if Raiden can not connect
 * :bug:`1911` The syncing message is now printed properly and does not repeat across the screen
 * :bug:`1899` Print proper error without throwing exception if no accounts are found in the keystore
+* :bug:`1975` Fix balance hash generation to match the logic that lives in ``TokenNetwork`` contract.
```
Comment (LefterisJP) [inline docs/changelog.rst]: Fix balance hash generation for zero transfers and empty locksroot

**Pull request merged**